### PR TITLE
Add travis-ci and code coverage plugins.

### PR DESCRIPTION
**Pull request by jeffmay**
@jeremyrsmith @travisbrown 

I'm having trouble hooking this up to codecov.io I think I need to be a member of the circe organization (instead of just the circe-yaml repo)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+scala:
+  - 2.11.8
+  - 2.10.6
+jdk:
+  - oraclejdk8
+
+script:
+  - sbt clean coverage test coverageReport
+after_success:
+  - sbt coveralls
```
Comment (jeffmay) [inline .travis.yml]: We should use codecov.io and switch this to match: https://github.com/codecov/example-scala/blob/master/.travis.yml

**Comment by travisbrown** [comment]
@jeffmay Just sent you an invite for the org. I'm headed out now but if that's not enough I can take a closer look later this evening.

**Comment by jeffmay** [comment]
@travisbrown awesome, got the invite. Looks like the admin needs to approve access to codecov.io

**Comment by travisbrown** [comment]
Okay, access granted!

**Pull request merged**